src/test/java/org/testng/AssertTest.java

```diff
@@ -381,6 +381,16 @@ public void testAssertSame() {
     Assert.assertSame(object2, object);
   }
 
+  @Test(description = "GITHUB-2490", expectedExceptions = AssertionError.class)

```

Comment:
We are not expecting `AssertionError` errors: `null` and `new BrokenEquals()` are not equals, which is right.

Reply:
Are you telling me to remove the expectedExceptions ? Without this, both the scripts are failing. To pass the scripts, I have added the expectedExceptions.

Reply:
Exactly. It is what is asked by the issue, right?

If the tests are failing, it only means it is not fixed.

Reply:
No, `new BrokenEquals()` and `null` *should* not be equals, but `new BrokenEquals().equals(null)` is `true`.
Therefore we are expecting an `AssertionError`.